client/src/main/java/io/pravega/client/stream/ReaderGroup.java

```diff
@@ -81,7 +82,18 @@
      * 
      * @param checkpoint The checkpoint to restore to.
      */
-    void resetReadersToCheckpoint(Checkpoint checkpoint);
```

Comment:
I would challenge the rationale for making a breaking change here.    The RG provides a checkpointing facility with familiar semantics of _checkpoint_ and _restore_, and `resetReadersToCheckpoint` is part of that facility.   It is a Good Thing that the checkpoint object encapsulates the RG state, and we should resist dumbing it down to mere cuts.   Consider that we may add time information to checkpoint state in the future.

I think that stream cuts are best treated as parameters during RG initialization.   We have a pattern already for that, the `ReaderGroupConfig`.   Notice that it has a `startingTime` parameter.   This new `resetReaders` method is inconsistent with that.  

I would prefer that start and end cuts be added to `ReaderGroupConfig`.

Reply:
>I would challenge the rationale for making a breaking change here.

Just wanted to highlight that the api to "reset readers to a Checkpoint" has not been removed. 
With this change the ReaderGroup has both the APIs.
```
void resetReaders(Checkpoint checkpoint); // reset readers to a Checkpoint.
void resetReaders(Map<Stream, StreamCut> streamCuts); // reset readers to the provided streamCuts.
```

Reply:
Update: 
Based on the current discussions with @fpj and @tkaitchuck, presence of `startingTime` in `ReaderGroupConfig` will be refactored based on the outcome of `Design Event time` story (this is currently under progress). This will be handled as part of a separate PR.

Reply:
I still think it makes most sense for the 'initial position' to be conveyed as a property of the `ReaderGroupConfig`.    If we also want to provide a seek API, then `resetReaders` would better be named `seek` in my opinion.   Note that we already have an `updateConfig` API for resetting the group to a new configuration (which would include a new initial position).

I also still think that renaming the `resetReadersToCheckpoint` method is an unnecessary change.   Why are you trying to conflate the checkpoint functionality with seek functionality???

Reply:
I think of 'seek' as doing something different from what this method does. This method causes all readers to irrecoverably fail so they are forced to reinitialize from the provided position. When I think of 'seek' it seems to imply that the existing reader would simply move forward in the stream.

Reply:
@EronWright 
> I still think it makes most sense for the 'initial position' to be conveyed as a property of the ReaderGroupConfig.

I can see this API being used in two ways:
- An existing reader group at a given time wants to resume reading from a given position. For this case, a call to seek/reset makes sense because it is an ongoing reader group.
- A new group is being created and configured to start reading from a given position. In this case, passing the position as part of the configuration makes sense.

We could conflate the two points and pass everything through configuration. Our way of configuring enables changes on-the-fly. I tend to think of configuration as values that are mostly static, e.g., the streams that a reader group reads from.

> I also still think that renaming the resetReadersToCheckpoint method is an unnecessary change.

I think I made that suggestion. The reason I suggested it is that the reset functionality is very similar in both cases, and when I see API calls like that, I prefer to disambiguate using the parameters in the signature. It makes the method calls shorter, and I don't feel it creates any confusion.

@tkaitchuck 
> I think of 'seek' as doing something different from what this method does.

It sounds right, although I could argue that the reinitialization prevents us from having a true seek. The reason I suggested this to be a seek functionality is mainly because I'm after the ability of telling a reader group to resume reading from arbitrary positions over time, just like we would do with seeking to specific positions in a file.